Sources/ClientRuntime/Networking/Http/AsyncThrowingStreamReader.swift

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AwsCommonRuntimeKit
+
+/// A `StreamReader` that reads `AsyncThrowingStream<Data, Error>`
+/// Event Stream messages are sent as a series of `Data` chunks
+public class AsyncThrowingStreamReader: StreamReader {
```

Comment:
Can we provide more descriptive error messages for each `fatalError` in this class? 

Additionally, can we add comments to methods that have _real_ implementations? (No need to add docs for methods that simply fatal error)

Reply:
done.